tests/prompt_strategies/test_chat_templates_advanced.py

```diff
@@ -912,5 +912,84 @@ def verify_labels(labels_span, should_train, context_message):
+        LOG.info("Testing combined optional fields with template")
+
+        # Template that includes both tool_calls and thoughts
+        template = """{% for message in messages %}
+{{'<|im_start|>'}}{{ message['role'] }}
+{% if message['thoughts'] is defined %}[Thoughts: {{ message['thoughts'] }}]{% endif %}
+{{ message['content'] }}{{'<|im_end|>'}}
+{% endfor %}"""
+
+        tokenizer = deepcopy(request.getfixturevalue("llama3_tokenizer"))
+
+        test_data = {
+            "messages": [
+                {"role": "system", "content": "You are an AI assistant."},
+                {"role": "user", "content": "What is the temperature in Paris?"},
+                {
+                    "role": "assistant",
+                    "content": "Let me help with that calculation",
+                    "thoughts": "We should take care to convert the temperature to Fahrenheit",
+                },
+            ]
+        }
+
+        strategy = ChatTemplateStrategy(
+            ChatTemplatePrompter(
+                tokenizer,
+                chat_template=template,
+                message_field_role="role",
+                message_field_content="content",
+                optional_message_fields=["thoughts"],
+            ),
+            tokenizer=tokenizer,
+            train_on_inputs=False,
+            sequence_len=512,
+            roles_to_train=["assistant"],
+        )
+
+        res = strategy.tokenize_prompt(test_data)
+        turns = strategy.get_conversation_thread(test_data)
+        labels = res["labels"]
+        input_ids = res["input_ids"]
+
+        LOG.debug(f"Labels: {labels}")
+        LOG.debug(f"Input IDs: {input_ids}")
+
+        # Verify both optional fields are in the tokenized output
+        decoded_output = tokenizer.decode(input_ids)
+
+        LOG.debug(f"Decoded output: {decoded_output}")
+        assert (
+            "[Thoughts: We should take care to convert the temperature to Fahrenheit]"
+            in decoded_output
+        ), "Thoughts not found in output"
+
+        start_idx, end_idx = strategy.find_turn(turns=turns, turn_idx=0)
+        turn_labels = labels[start_idx:end_idx]
+        assert all(
+            label == IGNORE_TOKEN_ID for label in turn_labels
+        ), "Expected system message content to be not to be trained on"
+
+        start_idx, end_idx = strategy.find_turn(turns=turns, turn_idx=1)
+        turn_labels = labels[start_idx:end_idx]
+        assert all(
+            label == IGNORE_TOKEN_ID for label in turn_labels
+        ), "Expected user message content to be not to be trained on"
+
+        # Verify all content is properly labeled for assistant message
+        start_idx, end_idx = strategy.find_turn(turns=turns, turn_idx=2)
+        turn_labels = labels[start_idx:end_idx]
+        assert not all(
+            label == IGNORE_TOKEN_ID for label in turn_labels
+        ), "Expected assistant message content to be trained on"
```

Comment:
```suggestion
        assert all(
            label != IGNORE_TOKEN_ID for label in turn_labels
        ), "Expected assistant message content to be trained on"
```

I think this is the condition we want?